__tests__/integration/sns/custom-mapping-template/tests.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+const { default: awsTesting } = require('aws-testing-library/lib/chai')
+const chai = require('chai')
+chai.use(awsTesting)
+const { expect } = chai
+
+const fetch = require('node-fetch')
+const { deployWithRandomStage, removeService } = require('../../../utils')
+
+describe('Single SNS Proxy Integration Test', () => {
+  let endpoint
+  let stage
+  let region
+  let queueUrl
+  const config = '__tests__/integration/sns/custom-mapping-template/service/serverless.yml'
+
+  beforeAll(async () => {
+    ;({
```

Comment:
why the leading `;` here?

Reply:
is this a js thing? looks funky lol

Reply:
It's a js thing since the auto formatting for the repo doesn't have trailing `;`.
I would rather just add `;` at the end of each line

Reply:
ok, what's wrong with this?

```
beforeAll(async () => {
  { stage, region, endpoint, outputs: { SqsQueueUrl: queueUrl } } = await deployWithRandomStage(config)
})
```
Does the code you have there behalve differently?

Reply:
That's actually the main reason I didn't use that method before, but when you have many variables you need to destruct it is still nicer that multiple lines

Reply:
You can't destruct into existing variables without wrapping the line with `()`

Reply:
Sorry, I'm confused. So the snippet I had does the same thing? If that's the case, why the leading `;`?

Reply:
ah, do you mean, have them on separate lines, like this?

```
beforeAll(async () => {
  { stage, 
    region, 
    endpoint, 
    outputs: { SqsQueueUrl: queueUrl } 
  } = await deployWithRandomStage(config)
})
```
is this not valid?

Reply:
We could just change `prettier.config.js` to use `semi: true`, but we can do it in another maintenance pr.

Reply:
ah! that's right, thanks for reminding me ;-)

> You can't destruct into existing variables without wrapping the line with `()`.
> You get a syntax error

and if you have `()` wrapping the whole thing then you also get a syntax error without the leading `;`.. right I think I get it now

Reply:
Sorry I'll try to be more clear.

You can do:

```
let { a } = result
```

but not

```
let a

{ a } = result
```

You can do:

```
let a;

({ a } = result);
```

or
 
```
let a
;({ a } = result))
```
or

```
let a 

a = result.a
```

Reply:
Ok, we can discuss about the config in another issue. Actually, either one is fine for me.

Reply:
cheers, thanks for the detailed explanation 👍